### PR TITLE
Implements Skeleton

### PR DESCRIPTION
**Pull request by dzucconi**
Continuing to pull some things off that `ResponsiveImage` PR. Here's `Skeleton` — which I currently have a use-case for in Force.

<details>
  <summary>📦 Published PR as canary version: <code>11.12.1-canary.741.12321.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```sh
  npm install @artsy/palette@11.12.1-canary.741.12321.0
  # or 
  yarn add @artsy/palette@11.12.1-canary.741.12321.0
  ```
</details>

### CODE REVIEW

File: packages/palette/src/elements/Skeleton/Skeleton.story.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+storiesOf("Components/Skeleton", module).add("Basic", () => {
+  return (
+    <>
+      <Skeleton m={2} width={400} height={300} />
+      <Skeleton m={2} width={400} height={300} done />
+      <Skeleton m={2} width={400} height={300} borderRadius="1em" />
+    </>
+  )
+})
```
Comment (zephraph) [inline packages/palette/src/elements/Skeleton/Skeleton.story.tsx:13]: Won't this always cause the visual tests to fail?

Comment (dzucconi) [reply packages/palette/src/elements/Skeleton/Skeleton.story.tsx:13]: Ah, good catch. I'll disable animation here for Chromatic.

**Comment by dzucconi** [comment]
OK so @zephraph

Re: https://www.chromatic.com/docs/animations
> Chromatic will pause CSS animations and reset them to their beginning state.

Since these are CSS animations, I think they should be fine. But I took the liberty to improve this a bit based on my current use-case so maybe take another look

File: packages/palette/src/utils/splitProps.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        leftProps[key] = props[key]
+        continue
+      }
+
+      rightProps[key] = props[key]
+    }
+
+    return [leftProps, rightProps]
+  }
+}
```
Comment (dzucconi) [inline packages/palette/src/utils/splitProps.ts:26]: USEFUL

File: packages/palette/src/elements/Text/Text.tsx [hunk 3]
```diff
@@ -12,10 +12,12 @@ export const overflowEllipsisMixin = css`
   text-overflow: ellipsis;
 `
 
+/** TextProps */
+export type TextProps = BaseTextProps &
+  BoxProps & { overflowEllipsis?: boolean }
+
```
Comment (dzucconi) [inline packages/palette/src/elements/Text/Text.tsx:18]: Very much looking forward to getting rid of the `.shared`

**Comment by dzucconi** [comment]
Going to just close/re-open to bump a canary because I want to be using it now.

**Comment by dzucconi** [comment]
OK... that didn't work.

**Comment by dzucconi** [comment]
OK, pushing empty commit, then re-pushing with empty commit pulled off 🗡️

File: packages/palette/src/utils/splitProps.ts (hunk 2 above)
Comment (zephraph) [reply packages/palette/src/utils/splitProps.ts:26]: Ahh, so the generic method of the splitBoxProps. I like.

**Comment by dzucconi** [comment]
BTW, this is why I want the `SkeletonText` — Let's us make very precise placeholders without too much tinkering:

placeholder

**Comment by dzucconi** [comment]
Trying to figure out why the typings in the canary differ from what I'm seeing in Storybook:

File: packages/palette/src/elements/Skeleton/Skeleton.story.tsx [hunk 4]
```diff
@@ -1,13 +1,53 @@
...
+          loading
+        </SkeletonText>
+
+        <SkeletonText variant="mediumText" m={1} borderRadius={4}>
+          still waiting...
+        </SkeletonText>
+
+        <SkeletonText variant="title" m={1} borderRadius={4}>
+          please wait
+        </SkeletonText>
```
Comment (damassi) [inline packages/palette/src/elements/Skeleton/Skeleton.story.tsx:29]: Cool idea 👍

**Comment by zephraph** [comment]
This is looking good to me! 👍

**Comment by artsyit** [comment]
:rocket: PR was released in [`v11.13.0`](https://github.com/artsy/palette/releases/tag/v11.13.0) :rocket:

**Pull request merged**